src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/RememberMeFactory.php

```diff
@@ -68,8 +131,29 @@ public function create(ContainerBuilder $container, $id, $config, $userProvider,
         // remember-me options
         $rememberMeServices->replaceArgument(3, array_intersect_key($config, $this->options));
 
-        // attach to remember-me aware listeners
+        $rememberMeServices->replaceArgument(0, $this->findUserProviders($container, $id, $config));
+    }
+
+    private function findUserProviders(ContainerBuilder $container, $id, array $config)
+    {
+        $userProviders = $this->findAndAttachToRememberMeAwareListeners($container, $id, $config);
+
+        if ($config['user_providers']) {
```

Comment:
BTW, is it OK that `user_providers` from `$config` override remember me aware listeners? Or is that a bug?